subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/AbstractWritableResultsStore.java

```diff
@@ -0,0 +1,103 @@
+        "(\n" +
+        "   select\n" +
+        "   testClass,\n" +
+        "   testId,\n" +
+        "   testProject,\n" +
+        "   operatingSystem,\n" +
+        "   max(id) as lastId\n" +
+        "   from testExecution\n" +
+        "   where startTime > ?\n" +
+        "   and (channel in (?, ?))\n" +
+        "   and testProject is not null\n" +
+        "   group by testClass,\n" +
+        "   testId,\n" +
+        "   testProject\n" +
+        ")\n" +
+        "select\n" +
+        "last.testClass,\n" +
+        "last.testId,\n" +
+        "last.testProject,\n" +
+        "testExecution.startTime,\n" +
+        "testExecution.endTime\n" +
+        "from last\n" +
+        "join testExecution on testExecution.id = last.lastId\n" +
+        "order by last.testClass,last.testId,last.testProject";
+
+    @Override
+    public Map<PerformanceExperiment, Long> getEstimatedExperimentTimes(OperatingSystem operatingSystem) {
+        return withConnection("load estimated runtimes", connection -> {
+            Timestamp since = Timestamp.valueOf(LocalDateTime.now().minusDays(LATEST_EXECUTION_TIMES_DAYS));
+            ImmutableMap.Builder<PerformanceExperiment, Long> builder = ImmutableMap.builder();
+            try (PreparedStatement statement = connection.prepareStatement(SELECT_LATEST_EXECUTION_TIMES)) {
+                statement.setTimestamp(1, since);
+                List<String> channels = ImmutableList.of("commits", "experiments").stream()
+                    .map(channel -> channel + operatingSystem.getChannelSuffix() + "-master")
+                    .collect(Collectors.toList());
+                statement.setString(2, channels.get(0));
+                statement.setString(3, channels.get(1));
+                try (ResultSet experimentTimes = statement.executeQuery()) {
+                    while (experimentTimes.next()) {
+                        String testClass = experimentTimes.getString(1);
+                        String testName = experimentTimes.getString(2);
+                        String testProject = experimentTimes.getString(3);
+                        long startTime = experimentTimes.getTimestamp(4).getTime();
+                        long endTime = experimentTimes.getTimestamp(5).getTime();
+                        if (testProject != null && testClass != null) {
+                            PerformanceExperiment performanceExperiment = new PerformanceExperiment(testProject, new PerformanceScenario(testClass, testName));
+                            builder.put(performanceExperiment, endTime - startTime);
+                        }
+                    }
+                    return builder.build();
+                }
+            }
+        });
+    }
+
+    protected <RESULT> RESULT withConnection(String actionName, ConnectionAction<RESULT> action) {
+        return db.withConnection(actionName, action);
+    }
+
+
+        @Override
```

Comment:
Nit: seems like incorrect indent.

Reply:
Fixed in https://github.com/gradle/gradle/pull/14672